KUBE-141: Remove stale mms.https.PEMKeyFile from Ops Manager config when TLS is disabled (#1389)
mongodb/mongodb-kubernetes: changelog/20260716_fix_ops_manager_stale_properties_after_disabling_tls.md

---
kind: fix
date: 2026-07-16
---

> * **MongoDBOpsManager**: Fixed Ops Manager failing to start after TLS is disabled when the configuration directory is persisted (for example a PVC mounted over `/mongodb-ops-manager/conf`). Properties the operator no longer sets, such as `mms.https.PEMKeyFile`, are now removed from `conf-mms.properties` on container start instead of lingering forever.

Reviewer: ```suggestion
* **MongoDBOpsManager**: Fixed the issue where Ops Manager failed to start after disabling the TLS when the configuration directory is persisted (for example a PVC mounted over `/mongodb-ops-manager/conf`).
```
